### DISCUSSION TITLE
Retrieve decoded size from snappy encoded data without decoding

### DISCUSSION THREAD

**Discussion by tanner-bruce** [klauspost/compress#488] [Q&A] [answered]
In grpc-go it's possible to help optimize the allocations during message decoding by providing the [decoded size up front](https://github.com/grpc/grpc-go/blob/master/encoding/encoding.go#L52-L55). Is it possible to do this with this library?

**Comment by klauspost** [comment, marked as answer]
For snappy blocks you can use [DecodedLen](https://pkg.go.dev/github.com/klauspost/compress@v1.14.2/s2#DecodedLen).

For streams you can use [IndexStream](https://pkg.go.dev/github.com/klauspost/compress@v1.14.2/s2#IndexStream) which you can [Load](https://pkg.go.dev/github.com/klauspost/compress@v1.14.2/s2#Index.Load) to get the [TotalUncompressed](https://pkg.go.dev/github.com/klauspost/compress@v1.14.2/s2#Index) size.

**Reply by tanner-bruce** [reply]
Awesome, thanks for the super quick response! I'll give that a try.
